shared/test/test_assets.rb

```diff
@@ -454,4 +446,21 @@ def post_file(assets, channel_id, filename, contents, content_type)
     response = post(assets, channel_id, file)
     [response, tmp_filename]
   end
+
+  def assert_custom_metric(index, metric_type, length_msg = nil)
+    # Filter out metrics from other test cases.
+    metrics = NewRelic::Agent.get_metrics %r{^Custom/FilesApi}
+    length_msg ||= "custom metrics recorded: #{index}"
+    assert_equal index, metrics.length, length_msg
+    last_metric = metrics.last
+    assert_equal "Custom/FilesApi/#{metric_type}_assets", last_metric.first, "#{metric_type} metric recorded"
+    assert last_metric.last == 1, "#{metric_type} metric value 1"
```

Comment:
This is a simple counter which I want to increase by 1 every time the event happens. I've extracted `expected_value` as a parameter.